src/ngraph/runtime/cpu/pass/cpu_memory_assignment.cpp

```diff
@@ -84,18 +84,18 @@ void runtime::cpu::pass::CPUMemoryAssignment::process_in_place_concat(
                     if (found_last_concat)
                     {
                         auto output_tensor = &concat->get_output_tensor();
-                        NGRAPH_ASSERT(m_tensor_to_key_map.find(output_tensor) !=
-                                      m_tensor_to_key_map.end());
-                        auto output_key = m_tensor_to_key_map[output_tensor];
+                        NGRAPH_ASSERT(m_tensor_to_bufferID.find(output_tensor) !=
+                                      m_tensor_to_bufferID.end());
+                        auto output_key = m_tensor_to_bufferID[output_tensor];
```

Comment:
This comment is unrelated to this change set but we should address it separately. I can see there are plenty of NGRAPH_ASSERT(map.find(X)) + map[X] pattern in this file. This pattern is problematic because:

1. We are doing two loop-ups when NGRAPH_ASSERT is enabled (inefficient).
2. When NGRAPH_ASSERT is disabled, map[X] won't fail, returning a newly created value. This may lead to unexpected behavior if the key is not in the map.

I would suggest doing:
```
auto output_buffer_it = m_tensor_to_bufferID.find(output_tensor);
NGRAPH_ASSERT(output_buffer_it != m_tensor_to_bufferID.end());
auto output_key = *output_buffer_it;
```